### PR TITLE
Update torch-xpu-ops commit pin

### PR DESCRIPTION
**Pull request by CuiYifeng**
Update the torch-xpu-ops commit to [intel/torch-xpu-ops@d87e68](https://github.com/intel/torch-xpu-ops/commit/d87e686f7cc6987aad3568acf6f362ee9378e23a), includes:

Build system:

- Replace enable_if SFINAE with C++20 requires-clauses

Kernel optimization/enabling/fixes:

- Replace shift_group_left with reduce_over_group in Reduce Sum for better portability
- Make deterministic tie-breaking stable in XPU topk kernel
- Enable dropout for FlashAttention fwd/bwd (reopened)
- Use if constexpr for compile-time-constant branches in softmax/Adam kernels
- Add float8 support and migrate to AT_DISPATCH_V2 in foreach copy kernels
- Enable bComplex32 dispatch for XPU unary kernels, unary geometric ops, where kernel, and arithmetic/compare operators

### CODE REVIEW

File: .ci/pytorch/build.sh [hunk 1]
```diff
@@ -185,6 +185,18 @@ if [[ "$BUILD_ENVIRONMENT" == *cuda* ]] && echo "${TORCH_CUDA_ARCH_LIST}" | tr '
...
+# XPU build sycltla require large amounts of meomory to build and will OOM
+if [[ "$BUILD_ENVIRONMENT" == *xpu* ]]; then
+  J=2  # default to 2 jobs
+  case "$RUNNER" in
+    linux.c7i.12xlarge)
+      J=24
+      ;;
+  esac
+  echo "Building sycltla with job limit $J"
+  export BUILD_CUSTOM_STEP="ninja -C build torch-xpu-ops-sycltla -j ${J}"
```
Comment (LuFinch) [inline .ci/pytorch/build.sh]: `export BUILD_CUSTOM_STEP="ninja -C build -t targets | grep -q torch-xpu-ops-sycltla && ninja -C build torch-xpu-ops-sycltla -j ${J} || echo 'sycltla target not found, skipping'"` to skip build torch-xpu-ops-sycltla when gcc < 13 or windows

Comment (LuFinch) [reply .ci/pytorch/build.sh]: 24 might be to high now. let's try smaller number

**Review by EikanWang — APPROVED** [review]

**Comment by CuiYifeng** [comment]
These two failed cases are irrelevant to this PR:
test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleGpu::test_copy_non_blocking_is_pinned_xpu
test/inductor/test_aot_inductor.py::AOTInductorTestDualWrapper::test_copy_non_blocking_is_pinned_xpu
